### DISCUSSION TITLE
javascript library shipped with AE-govDesign System

### DISCUSSION THREAD

**Discussion by usmanehsan0613** [TDRA-ae/aegov-dls#65] [General]
Hi there, 

as per the documentation , it says by default the javascript libraries are included in " if the bundled JavaScript library of AEGov-Design-System is utilized." 

where it does gets installed , i have installed the three plugins but modals and accordian doesnot work. I want to use exactly what TDRA has provided rather writing my own addOns

kindly advise.

**Comment by moiz-conceptlz** [comment]
Hello @usmanehsan0613 

Thank you for reaching out!

The aegov-scripts.js file is available in the following path within the `node_modules` directory:
`node_modules/@aegov/design-system/dist/aegov-scripts.js` 

This file is a bundled script that includes all the plugins provided by the Design System, including modals and accordions. You can use it directly without needing to write your own add-ons.

## How to import and use the script

Assuming you are using a build system - such as webpack, vite or gulp, you would most likely writing some custom JavaScript yourself for your project. Hence, the best approach is to use `import '@aegov/design-system/dist/aegov-scripts.js'; ` at the very top of your JS file and then compile your code. This way, (for example; with Webpack) the script fill will be read and the output JavaScript will include the code.

This is the most appropriate method, because any update to the JavaScript file itself with an update in the plugin will simply compile wit your code.

Assuming you are not using a build system - then copy the file from the vendor folder and place it in your directory. You can then use this file as it is from your directory. This is a Vanilla JS coded file with no dependency on any library. 

This is not the ideal approach, because any update to the plugin code would need you to copy the file again.

Make sure the required HTML structure and classes for the modals and accordions align with what is specified in the documentation. If the issue persists, check your browser console for errors or confirm that the library and its dependencies are installed correctly.

Let me know if you need further assistance!
